sycl/test/fpga_tests/fpga_pipes.cpp

```diff
@@ -0,0 +1,207 @@
+
+      cgh.parallel_for<class boo_nb>(range<1> { dataSize }, [=](id<1> idx) {
+        bool SuccessCode;
+        while (!SuccessCode)
+          pipe<class some_pipe, int, 1>::write(read_acc[idx], SuccessCode);
+      });
+    });
+
+    buffer<int, 1> writeBuf (data, range<1>(dataSize));
+    Queue.submit([&](handler& cgh) {
+      auto write_acc = writeBuf.get_access<access::mode::write>(cgh);
+
+      cgh.parallel_for<class zoo_nb>(range<1> { dataSize }, [=](id<1> idx) {
+        bool SuccessCode;
+        while (!SuccessCode)
+          write_acc[idx] = pipe<class some_pipe, int, 1>::read(SuccessCode);
+      });
+    });
+
+    auto readHostBuffer = writeBuf.get_access<access::mode::read>();
+    for (size_t i = 0; i != dataSize; ++i) {
+      if (readHostBuffer[i] != 42) {
+        std::cout << "Result mismatches " << readHostBuffer[i]
+                  << " Vs expected " << 42 << " for index " << i << std::endl;
+
+        return -1;
+      }
+    }
+  }
+
+  {
+    // Test for simple non-blocking pipes created with namespaces set
+    queue Queue;
+
+    buffer<int, 1> readBuf (data, range<1>(dataSize));
+    Queue.submit([&](handler& cgh) {
+      auto read_acc = readBuf.get_access<access::mode::read>(cgh);
+
+      cgh.parallel_for<class foo_ns>(range<1> { dataSize }, [=](id<1> idx) {
+        bool SuccessCode;
+        while (!SuccessCode)
+          pipe<class some::pipe, int, 1>::write(read_acc[idx], SuccessCode);
+      });
+    });
+
+    buffer<int, 1> writeBuf (data, range<1>(dataSize));
+    Queue.submit([&](handler& cgh) {
+      auto write_acc = writeBuf.get_access<access::mode::write>(cgh);
+
+      cgh.parallel_for<class boo_ns>(range<1> { dataSize }, [=](id<1> idx) {
+        bool SuccessCode;
+        while (!SuccessCode)
+          write_acc[idx] = pipe<class some::pipe, int, 1>::read(SuccessCode);
+      });
+    });
+
+      auto readHostBuffer = writeBuf.get_access<access::mode::read>();
+      for (size_t i = 0; i != dataSize; ++i) {
+        if (readHostBuffer[i] != 42) {
+          std::cout << "Result mismatches " << readHostBuffer[i]
+                    << " Vs expected " << 42 << " for index " << i << std::endl;
+
+        return -1;
+      }
+    }
+  }
+
+  {
+    // Test for forward declared pipes
+    queue Queue;
+    class pipe_type_for_lambdas;
+
+    buffer<int, 1> readBuf (data, range<1>(dataSize));
+    Queue.submit([&](handler& cgh) {
+      auto read_acc = readBuf.get_access<access::mode::read>(cgh);
+
+      cgh.parallel_for<class foo_la>(range<1> { dataSize }, [=](id<1> idx) {
+        bool SuccessCode;
+        while (!SuccessCode)
+          pipe<class pipe_type_for_lambdas, int>::write(read_acc[idx],
+                                                      SuccessCode);
+      });
+    });
+
+    buffer<int, 1> writeBuf (data, range<1>(dataSize));
+    Queue.submit([&](handler& cgh) {
+      auto write_acc = writeBuf.get_access<access::mode::write>(cgh);
```

Comment:
`write_acc` is not used here, is that intentional?

Reply:
To remove, thanks!